src/main/resources/templates/java/test/testFiles/structural/AttributeTest.java

```diff
@@ -1,34 +1,21 @@
 package ${packageName};
 
-import static org.junit.jupiter.api.Assertions.fail;
-import static org.junit.jupiter.api.DynamicContainer.dynamicContainer;
-import static org.junit.jupiter.api.DynamicTest.dynamicTest;
-
-import java.lang.annotation.Annotation;
-import java.lang.reflect.Field;
-import java.lang.reflect.Modifier;
-import java.lang.reflect.ParameterizedType;
-import java.lang.reflect.Type;
-import java.net.URI;
 import java.net.URISyntaxException;
-import java.util.ArrayList;
-import java.util.List;
 
-import org.json.JSONArray;
-import org.json.JSONObject;
 import org.junit.jupiter.api.DynamicContainer;
-import org.junit.jupiter.api.DynamicNode;
 import org.junit.jupiter.api.TestFactory;
 import org.junit.jupiter.api.Timeout;
 
+import de.tum.in.test.api.structural.AttributeTestProvider;
+
 /**
  * @author Stephan Krusche (krusche@in.tum.de)
- * @version 4.0 (27.10.2020)
+ * @version 5.0 (11.11.2020)
  * <br><br>
  * This test evaluates if the specified attributes in the structure oracle are correctly implemented with the expected type, visibility modifiers and annotations,
  * based on its definition in the structure oracle (test.json).
  */
-public class AttributeTest extends StructuralTest {
+public class AttributeTest extends AttributeTestProvider {
```

Comment:
It would be a good idea to add the AJTS annotations here as well. (Can be the same as in the behavior)

We currently initialize the classes AFAIK.

Reply:
The same holds for the other three.